velox/parse/DuckLogicalOperator.h

```diff
@@ -0,0 +1,198 @@
+// copy of fmt/format.h.
+
+/*
+Copyright 2018-2022 Stichting DuckDB Foundation
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#include "velox/external/duckdb/duckdb.hpp"
+
+namespace duckdb {
+
+//! LogicalDummyScan represents a dummy scan returning a single row
+class LogicalDummyScan : public LogicalOperator {
+ public:
+  explicit LogicalDummyScan(idx_t table_index)
+      : LogicalOperator(LogicalOperatorType::LOGICAL_DUMMY_SCAN),
+        table_index(table_index) {}
+
+  idx_t table_index;
+
+ public:
+  vector<ColumnBinding> GetColumnBindings() override {
+    return {ColumnBinding(table_index, 0)};
+  }
+
+  idx_t EstimateCardinality(ClientContext& context) override {
+    return 1;
+  }
+
+ protected:
+  void ResolveTypes() override {
+    if (types.size() == 0) {
+      types.emplace_back(LogicalType::INTEGER);
+    }
+  }
+};
+
+//! LogicalGet represents a scan operation from a data source
+class LogicalGet : public LogicalOperator {
+ public:
+  LogicalGet(
+      idx_t table_index,
+      TableFunction function,
+      unique_ptr<FunctionData> bind_data,
+      vector<LogicalType> returned_types,
+      vector<string> returned_names);
+
+  //! The table index in the current bind context
+  idx_t table_index;
+  //! The function that is called
+  TableFunction function;
+  //! The bind data of the function
+  unique_ptr<FunctionData> bind_data;
+  //! The types of ALL columns that can be returned by the table function
+  vector<LogicalType> returned_types;
+  //! The names of ALL columns that can be returned by the table function
+  vector<string> names;
+  //! Bound column IDs
+  vector<column_t> column_ids;
+  //! Filters pushed down for table scan
+  TableFilterSet table_filters;
+
+  string GetName() const override;
+  string ParamsToString() const override;
+  //! Returns the underlying table that is being scanned, or nullptr if there is
+  //! none
+  TableCatalogEntry* GetTable() const;
+
+ public:
```

Comment:
public is restated. This line can be removed.

See this as a pattern in the rest of the classes too. Did you do so deliberately ?

Reply:
This is a copy-paste from duckdb-internal.hpp. Hence, cannot modify.